File: java/src/main/java/com/carepet/model/SensorAvg.java
package com.carepet.model;

import com.datastax.oss.driver.api.mapper.annotations.ClusteringColumn;
import com.datastax.oss.driver.api.mapper.annotations.CqlName;
import com.datastax.oss.driver.api.mapper.annotations.Entity;
import com.datastax.oss.driver.api.mapper.annotations.PartitionKey;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.time.LocalDate;
import java.util.UUID;

@Entity
@CqlName("sensor_avg")
public class SensorAvg {
    @PartitionKey
    @JsonProperty("sensor_id")
    private UUID sensorId;

    @ClusteringColumn(0)
    private LocalDate date;

    @ClusteringColumn(1)
    private int hour;

    private float value;

    public SensorAvg() {}

    public SensorAvg(UUID sensorId, LocalDate date, int hour, float value) {
        this.sensorId = sensorId;
        this.date = date;
        this.hour = hour;
        this.value = value;
    }

    public UUID getSensorId() {
        return sensorId;
    }

    public void setSensorId(UUID sensorId) {
        this.sensorId = sensorId;
    }

    public LocalDate getDate() {
        return date;
    }

    public void setDate(LocalDate date) {
        this.date = date;
    }

    public int getHour() {
        return hour;
    }

    public void setHour(int hour) {
        this.hour = hour;
    }

    public float getValue() {
        return value;
    }

    public void setValue(float value) {
        this.value = value;
    }

    @Override
    public String toString() {
        return "SensorAvg{" +
                "sensorId=" + sensorId +
                ", date=" + date +
                ", hour=" + hour +
                ", value=" + value +
                '}';
    }
}

File: csharp/Model/SensorAvg.cs
using System;
using Cassandra.Mapping.Attributes;
using Newtonsoft.Json;

namespace CarePet.Model
{
    [Table(Name = "sensor_avg")]
    public class SensorAvg
    {
        [PartitionKey]
        [Column("sensor_id")]
        [JsonProperty("sensor_id")]
        public Guid SensorId { get; set; }

        [ClusteringKey(0)]
        [Column("date")]
        public DateTime Date { get; set; }

        [ClusteringKey(1)]
        [Column("hour")]
        public int Hour { get; set; }

        [Column("value")]
        public float Value { get; set; }

        public SensorAvg() { }

        public SensorAvg(Guid sensorId, DateTime date, int hour, float value)
        {
            SensorId = sensorId;
            Date = date;
            Hour = hour;
            Value = value;
        }

        public override string ToString()
        {
            return $"SensorAvg{{sensorId={SensorId}, date={Date}, hour={Hour}, value={Value}}}";
        }
    }
}
